options/posix/generic/net-if-stubs.cpp

```diff
@@ -36,7 +36,15 @@ struct if_nameindex *if_nameindex(void) {
 	return NULL;
 }
 
-unsigned int if_nametoindex(const char *) {
-	__ensure(!"Not implemented");
-	__builtin_unreachable();
+unsigned int if_nametoindex(const char *name) {
+	struct ifreq ifr;
+	int fd, r;
+
+	if((fd = socket(AF_UNIX, SOCK_DGRAM|SOCK_CLOEXEC, 0)) < 0) {
+		return 0;
+	}
+	strncpy(ifr.ifr_name, name, sizeof(ifr.ifr_name));
```

Comment:
might be worth memsetting ifreq to 0 and using sizeof - 1 to ensure a null terminator exist, though, whether that matters is OS specific (and should be a sysdep)